src/main/java/chessmaster/user/Player.java

```diff
@@ -0,0 +1,171 @@
+     * history, all current pieces, and colour
+     * of each player. It also contains functions to request input from the user for
+     * the next move and to execute
+     * that move.
+     * 
+     * @param colour The ChessPiece.Colour desired for this player.
+     */
+    public Player(int colour) {
+        this.moves = new ArrayList<>();
+        this.pieces = new ArrayList<>();
+        this.colour = colour;
+    }
+
+    /**
+     * Adds a given move into the Player's move history.
+     * 
+     * @param move The given move to be added to history.
+     */
+    public void addMove(Move move) {
+        this.moves.add(move);
+    }
+
+    /**
+     * Adds all the player's pieces to their Piece array. Run once during setup of
+     * the game.
+     * 
+     * @param board The new ChessBoard containing all 32 chess pieces.
+     */
+    public void initialisePieces(ChessBoard board) {
+        int row;
+        int col;
+        
+        if (this.colour == ChessPiece.BLACK) {
+            row = 6;
+        } else {
+            row = 0;
+        }
+
+        for (int tempRow = row; row < tempRow + 2; row++) {
+            for (col = 0; col < ChessBoard.SIZE; col++) {
+                try {
+                    ChessPiece piece = board.getPieceAtCoor(new Coordinate(col, row));
+                    this.pieces.add(piece);
+                } catch (NullPieceException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+    }
+
+    /**
+     * Prints out all the player's pieces including whether it has been captured or
+     * not.
+     * Used for debugging purposes only.
+     */
+    public void printAllPieces() {
+        for (ChessPiece p : pieces) {
+            System.out.println("Piece: " + p);
+            System.out.println("Colour: " + p.getColour());
+            System.out.println("Is captured: " + p.getCaptured());
+        }
+    }
+
+    public int getColour() {
+        return this.colour;
+    }
+
+    /**
+     * High-level function which calls necessary APIs to request user input and
+     * parse it into a Move object.
+     * 
+     * @param board The board which to make the move on.
+     * @return null if the user inputs "abort", an empty Move if an error occurred
+     *         during parsing, otherwise return the requested Move object.
+     */
+    public Move getNextMove(ChessBoard board) throws InvalidMoveException, NullPieceException, ParseCoordinateException{
+        String input = TextUI.getUserInput(); // Get user input
+        try {
+            Parser parser = new Parser();
+            Command command = parser.parseCommand(input, board);
+            if (command instanceof AbortCommand) {
+                return null;
+            }
+            return command.getMove();
+        
+        } catch (ParseCoordinateException | NullPieceException e) {
+            TextUI.printErrorMessage(e);
+        }
+        return new Move();
+    }
```

Comment:
is the catch clause intended to throw the same exception again after printing error message?